sched/scheduler/stateful_scheduler.go

```diff
@@ -99,25 +100,39 @@ type jobAddedMsg struct {
 	saga *saga.Saga
 }
 
-func (s *statefulScheduler) ScheduleJob(job sched.Job) error {
-
+func (s *statefulScheduler) ScheduleJob(jobDef sched.JobDefinition) (string, error) {
 	defer s.stats.Latency("schedJobLatency_ms").Time().Stop()
 	s.stats.Counter("schedJobRequests").Inc(1)
 
+	job := sched.Job{
+		Id:  generateJobId(),
+		Def: jobDef,
+	}
+
 	// Log StartSaga Message
 	// TODO: need to serialize job into binary and pass in here
 	// so we can recover the job in case of failure
 	sagaObj, err := s.sagaCoord.MakeSaga(job.Id, nil)
 	if err != nil {
-		return err
+		return "", err
 	}
 
 	s.addJobCh <- jobAddedMsg{
 		job:  job,
 		saga: sagaObj,
 	}
 
-	return nil
+	return job.Id, nil
+}
+
+// generates a jobId using a random uuid
+func generateJobId() string {
+	id, err := uuid.NewV4()
+	for err != nil {
```

Comment:
So this is an artifact of the golang [rand.Read API](https://golang.org/pkg/math/rand/#Read) which is called as part of uuid.NewV4()  

"Read generates len(p) random bytes from the default Source and writes them into p. It always returns len(p) and a nil error. Read, unlike the Rand.Read method, is safe for concurrent use."

So basically this should never return an error, but should they make a change to the API and it does?  I think its so Rand implements the [Reader interface](https://golang.org/pkg/io/#Reader) 

I'll add this in a comment since its super weird.  I also looked it up when initially implementing, and change the for loop